src/com/amazon/corretto/crypto/provider/EvpSignatureBase.java

```diff
@@ -36,17 +36,46 @@ abstract class EvpSignatureBase extends SignatureSpi {
     protected int keyUsageCount_ = 0;
     protected String algorithmName_ = null;
     protected PSSParameterSpec pssParams_ = null;
+    // While digest_ isn't needed for the all instances of this abstract class, it's cleaner to manage it with the
+    // rest of the signature parameter values.
+    protected long digest_ = 0; // Must be kept in sync with pssParams_ or main algorithm name.
+    protected long pssMgfMd_ = 0; // Must be kept in sync with pssParams_
+    protected int pssSaltLen_ = 0; // Must be kept in sync with pssParams_
 
     EvpSignatureBase(
             final AmazonCorrettoCryptoProvider provider,
             final EvpKeyType keyType,
-            final int paddingType
+            final int paddingType,
+            final long digest
     ) {
         provider_ = provider;
         keyType_ = keyType;
         paddingType_ = paddingType;
         if (paddingType_ == RSA_PKCS1_PSS_PADDING) {
-            pssParams_ = PSSParameterSpec.DEFAULT;
+            internalSetParams(PSSParameterSpec.DEFAULT);
+            // digest_ is set by internalSetParameters
+        } else {
+            internalSetParams(null);
+            // Overwrite the 0 set by internalSetParameters
+            digest_ = digest;
```

Comment:
don't we want this to be set to `0` if `paddingType_` isn't PSS? no other padding types take a digest.

Reply:
This is the digest used to hash the message directly. (e.g., SHA-256 for SHA256withRSA)

Reply:
ah, right.